lib/plugins/inflightRequestThrottle.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+var assert = require('assert-plus');
+var ServiceUnavailableError = require('restify-errors').ServiceUnavailableError;
+var defaultResponse = new ServiceUnavailableError('resource exhausted');
+
+/**
+ * inflightRequestThrottle
+ *
+ * @param {Object} opts configure this plugin
+ * @param {Number} opts.limit maximum number of simultanenous
+ *    connections the server will handle before returning an error
+ * @param {Error} opts.resp the error object to pass to res.send when the
+ *    maximum concurrency is exceeded
+ * @param {Number} opts.resp.statusCode the status code to return when the
+ *    maximum concurrency is exceeded
+ * @param {Function} opts.server.inflightRequests A function that, when
+ *    invoked, returns the number of requests currently being handled by the
+ *    server.
+ * @returns {Function} middleware to be registered on server.pre
+ */
+function inflightRequestThrottle (opts) {
```

Comment:
Does this class need a constructor? Generally, the pattern for plugins in restify has been to use a closure, and obviating the need for a constructor. Can we keep it consistent and do that here? We can of course, debate closure vs constructor trade offs at another point in time. Here's an example: https://github.com/restify/node-restify/blob/5.x/lib/plugins/metrics.js#L23